src/frontend/web_application/src/components/MultidimensionalPi/components/PiGraph/index.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { PropTypes } from 'react';
+import './style.scss';
+
+const Grid = ({ pi, angle, axeLength }) => {
+  const axeCoordinates = [];
+  const polygonPoints = [];
+  let count = 0;
+
+  pi.map((p) => {
+    const axeName = p.name;
+    const axeX = axeLength - (axeLength * Math.sin((count * Math.PI) / 180));
+    const axeY = axeLength - (axeLength * Math.cos((count * Math.PI) / 180));
+    axeCoordinates.push({ axeName, x: axeX, y: axeY });
+    polygonPoints.push(axeX, axeY);
+    count -= angle;
+  });
+
+  return (
+    <g className="m-pi-graph__grid">
+      <polygon className="m-pi-graph__outline" points={polygonPoints.join(' ')} />
+      {axeCoordinates.map(p =>
+        <line
+          key={p.axeName}
+          className="m-pi-graph__line"
+          id={p.axeName}
+          x1={axeLength}
+          y1={axeLength}
+          x2={p.x}
+          y2={p.y}
+        />
+      )}
+    </g>
+  );
+};
+
+Grid.propTypes = {
+  pi: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+  angle: PropTypes.number.isRequired,
+  axeLength: PropTypes.number.isRequired,
+};
+
+const Polygon = ({ pi, angle, axeLength }) => {
+  const polygonPoints = [];
+  let count = 0;
+  pi.map((p) => {
```

Comment:
Agree with @ziir, I don't know what's the better between maps or reduce in this case, it depends on the attr `point` of polygon.

To answer your specific question, `map` should be used to transform the items of an array so you must return a value:
e.g I have `[1, 2, 3]` and I want `['a', 'b', 'c']`.

What you have in mind I think is a `forEach`: do something for each items of an array:
e.g I have an empty value `foo`, a func `doTheThing` and an array `[1, 2, 3]` then I want foo increments each time the func is true:

```js
let foo = 0;
[1, 2, 3].forEach(item => {
  if (doTheThing(item)) { 
    foo++; 
  }
});
```

It's procedural programing (like `for(...)`). You can also use `reduce`, so you don't need to init `foo`, it's called functional programing.